metaserver/src/main/scala/scala/meta/languageserver/search/SymbolIndex.scala

```diff
@@ -1,223 +1,39 @@
-        // are typically created at the same time for each project/configuration
-        // combination. Duplicate tasks are expensive, for example we don't want
-        // to index the JDK twice on first startup.
-        indexedJars.computeIfAbsent(jar, _ => buf += jar)
-      }
-      val sourceJarsToIndex = buf.result()
-      // Acquire a lock on the leveldb cache only during indexing.
-      LevelDBMap.withDB(cacheDirectory.resolve("leveldb").toFile) { db =>
-        sourceJarsToIndex.foreach { path =>
-          logger.info(s"Indexing classpath entry $path...")
-          val database = db.getOrElseUpdate[AbsolutePath, Database](path, {
-            () =>
-              Mtags.indexDatabase(path :: Nil)
-          })
-          indexDatabase(database)
-        }
-      }
-      Effects.IndexSourcesClasspath
-    }
-  }
-
-  /** Register this Database to symbol indexer. */
-  def indexDatabase(document: s.Database): Effects.IndexSemanticdb = {
-    document.documents.foreach(indexDocument)
-    Effects.IndexSemanticdb
-  }
-
-  /**
-   *
-   * Register this Document to symbol indexer.
-   *
-   * Indexes definitions, denotations and references in this document.
-   *
-   * @param document Must respect the following conventions:
-   *                 - filename must be a URI
-   *                 - names must be sorted
-   */
-  def indexDocument(document: s.Document): Effects.IndexSemanticdb = {
-    val input = Input.VirtualFile(document.filename, document.contents)
-    // what do we put as the uri?
-    val uri = URI.create(document.filename)
-    documentIndex.putDocument(uri, document)
-    document.names.foreach {
-      // TODO(olafur) handle local symbols on the fly from a `Document` in go-to-definition
-      // local symbols don't need to be indexed globally, by skipping them we should
-      // def isLocalSymbol(sym: String): Boolean =
-      // !sym.endsWith(".") &&
-      //     !sym.endsWith("#") &&
-      //     !sym.endsWith(")")
-      // be able to minimize the size of the global index significantly.
-      //      case s.ResolvedName(_, sym, _) if isLocalSymbol(sym) => // Do nothing, local symbol.
-      case s.ResolvedName(Some(s.Position(start, end)), sym, true) =>
-        symbolIndexer.addDefinition(
-          sym,
-          i.Position(document.filename, Some(input.toIndexRange(start, end)))
-        )
-      case s.ResolvedName(Some(s.Position(start, end)), sym, false) =>
-        symbolIndexer.addReference(
-          document.filename,
-          input.toIndexRange(start, end),
-          sym
-        )
-      case _ =>
-    }
-    document.symbols.foreach {
-      case s.ResolvedSymbol(sym, Some(denot)) =>
-        symbolIndexer.addDenotation(
-          sym,
-          denot.flags,
-          denot.name,
-          denot.signature
-        )
-      case _ =>
-    }
-    Effects.IndexSemanticdb
-  }
-
+  ): Effects.IndexSourcesClasspath
+  def indexDatabase(document: s.Database): Effects.IndexSemanticdb
```

Comment:
Can we move the docstrings from the in-memory index to here?

Reply:
Done.